docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -271,68 +330,125 @@ snapshot_dump_path=""
-a predefined location. 
-
-On startup, if `local_sync` is set to `true`, CometBFT will try to look for an existing snapshot at the path
-given by the operator. The snapshots will be loaded and state restored as if they came from a peer in the current implementation. 
+We could expand the CLI interface to allow for additional operations, similar to the above provided CLI. However,
+as snapshot generation, dumping and loading a local snapshot would be built into CometBFT,
+ while the node is running, we do not need to rely on a CLI for this functionality.
 
 The `dump` command can be implemented in two ways:
-- Rely on the existing ABCI functions `ListSnapshots` and `LoadChunks` to retrieve the snapshots and chunks from a peer. 
-This approach requires no change to the current API and is easy to implement. It however involves more than one ABCI call and 
-data transafer. 
-- Extend `RequestListSnapshots` with a flag to indicate that we want an exportable snapshot format and extend `ResponseListSnapshot` to return a 
+
+1. Rely on the existing ABCI functions `ListSnapshots` and `LoadChunks` to retrieve the snapshots and chunks from a peer. 
+This approach requires no change to the current API and is easy to implement. Furthermore, CometBFT has complete control
+over the format of the exported snapshot. It does however involve more than one ABCI call and network data transfer. 
+2.  Extend `RequestListSnapshots` with a flag to indicate that we want an exportable snapshot format and extend `ResponseListSnapshot` to return a 
 path and format of the exported snapshots.
-- Add path parameters to the command, and include the path into `RequestListSnapshots` instructing the application to generate a snapshot export
-at a given location. 
+
+An improvement to the second option mentioned above would be to add path parameters to the command,
+ and include the path into `RequestListSnapshots` instructing the application to generate a snapshot export
+at a given location.
+
+A third option is the introduction of a new ABCI call: `ExportSnapshots`, which will have the same effect as option 2 above.
+
+
+#### *Automatic snapshot exporting*
+
+Applications generate snapshots in fixed time intervals. The application itself measures the time passed since the last snapshot,
+and CometBFT has no role in instructing the application when to take snapshots. 
+
+The State sync reactor currently retrieves a list of snapshots from a peer, who obtains these snapshots via ABCI (`RequestListSnapshots`). 
+
+Applications can thus themselves be in charge of dumping the snapshots into a given file format, the same way they generate snapshots.
+ If `auto_snapshot_dump` is true, 
+CometBFT instructs the application to export the snapshots periodically.
+
+An alternative solution is that CometBFT, itself, using the implementation of the `dump` command, whichever 
+is chosen at the time, creates or asks the application to create snapshot exports. 
```

Comment:
To export existing ones. Although the application can do things differently if we ask the app to export it. Otherwise, if comet is doing it, we are just asking the app for current snapshots.